From 4044f6b2bf7cb9980b0d7760f87e4937afc2c455 Mon Sep 17 00:00:00 2001
From: Jukka Laitinen <jukka.laitinen@intel.com>
Date: Fri, 25 Nov 2016 19:21:27 +0200
Subject: [PATCH] Add detecting intel board for atmel_mxt driver

This is needed for the atmel_mxt to load properly for 5xx

Signed-off-by: Jukka Laitinen <jukka.laitinen@intel.com>
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index 887020a..64b666b 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -2512,6 +2512,29 @@ static struct mxt_acpi_platform_data chromebook_platform_data[] = {
 	{ }
 };
 
+static unsigned int intel_tp_buttons[] = {
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_RESERVED,
+};
+
+static struct mxt_acpi_platform_data intel_platform_data[] = {
+	{
+		/* Touchpad */
+		.hid	= "ATML0000",
+		.pdata	= {
+			.t19_num_keys	= ARRAY_SIZE(intel_tp_buttons),
+			.t19_keymap	= intel_tp_buttons,
+		},
+	},
+	{
+		/* Touchscreen */
+		.hid	= "ATML0001",
+	},
+	{ }
+};
+
 static const struct dmi_system_id mxt_dmi_table[] = {
 	{
 		/* 2015 Google Pixel */
@@ -2530,6 +2553,14 @@ static const struct dmi_system_id mxt_dmi_table[] = {
 		},
 		.driver_data = chromebook_platform_data,
 	},
+	{
+		/* Intel boards, such as 5xx */
+		.ident = "Intel",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Intel"),
+		},
+		.driver_data = intel_platform_data,
+	},
 	{ }
 };
 
-- 
2.7.4

